### PR TITLE
Fix: Correct Pagination for User Table Filter - 1430

### LINKED ISSUE #1430: LCFS BUG - Pagination incorrect for User Table
* Login as IDIR user
* Go to Users Page
* Apply Filter

Expected:
* Pagination shows how many items match the filters

Actual:
* Pagination shows the total number / original number of items

Image

### PR DESCRIPTION
**Pull request by hamed-valiollahi**
This PR fixes the pagination to display the correct count of filtered items on the User Table.

Closes #1430

### CODE REVIEW

File: backend/lcfs/web/api/user/services.py [hunk 1]
```diff
@@ -125,16 +125,16 @@ async def get_all_users(self, pagination: PaginationRequestSchema) -> UsersSchem
...
                 total=total_count,
                 page=pagination.page,
                 size=pagination.size,
-                total_pages=math.ceil(total_count / pagination.size),
+                total_pages=(
+                    math.ceil(total_count / pagination.size) if total_count > 0 else 0
+                ),
             ),
-            users=users,
+            users=users if users else [],
```
Comment (areyeslo) [inline backend/lcfs/web/api/user/services.py:137]: Not need to add contidition, `get_users_paginated` is returning  [] if no users or array with UserProfile entities

File: backend/lcfs/web/api/user/services.py [hunk 2]
```diff
@@ -125,16 +125,16 @@ async def get_all_users(self, pagination: PaginationRequestSchema) -> UsersSchem
         Get all users
         """
         users, total_count = await self.repo.get_users_paginated(pagination=pagination)
-        if len(users) == 0:
-            raise DataNotFoundException("No users found")
```
Comment (areyeslo) [inline backend/lcfs/web/api/user/services.py:129]: Great you removed this! It was not allowing to return the pagination details.

**Comment by hamed-valiollahi** [comment]
@areyeslo Thank you for the code review! The requested change has been applied.

**Review by areyeslo — APPROVED** [review]
LGTM

**Pull request merged**